src/sentry/api/bases/project.py

```diff
@@ -120,8 +122,8 @@ def convert_args(self, request, organization_slug, project_slug, *args, **kwargs
                     organization__slug=organization_slug,
                     redirect_slug=project_slug
                 )
-
-                raise ResourceMoved(detail={'slug': redirect.project.slug})
+                new_url = request.path.replace('projects/%s/%s/' % (organization_slug, project_slug), 'projects/%s/%s/' % (organization_slug, redirect.project.slug))
```

Comment:
This is going to drop any query strings.

You can leverage `request.get_full_path()` https://docs.djangoproject.com/en/1.7/ref/request-response/#django.http.HttpRequest.get_full_path

Which also includes the querystring.

I'd also introduce a check for like, `if new_url != old_url` before triggering a redirect, otherwise, you'll end in a redirect loop. This would happen if say, a different url that didn't follow this pattern of `projects/%s/%s/` came through. I'd say we'd want to raise a 404 in that case instead of an endless loop of redirection.